Javascript logic

RoyNyaga opened the pull request:
All the javascript logic of this project is found in this branch. I made use of javascript ES6 modules and used webpack to bundle the modules together.

RoyNyaga committed:
created Tools module in index.js and included chnageTab function

RoyNyaga committed:
added arguments to loadMenuPage, loadHomepage, loadContactPage functions

RoyNyaga committed:
imported and exported dependencies

RoyNyaga committed:
added photos and fixed errors

RoyNyaga committed:
added webpack config file

RoyNyaga committed:
fixed more errors and added stickler.yml file

RoyNyaga committed:
fixed more errors

RoyNyaga committed:
added readme file, ignored webpacker.config.js file in stickler.yml file.

bolah2009 commented on README.md:
```diff
@@ -0,0 +1,13 @@
+
+**Authors**
+### [Nyaga Roy](https://github.com/RoyNyaga).
+
```
We can make the readme more descriptive by adding more details about the project like screenshot and live link, eg: https://raw.githack.com/RoyNyaga/microverse_restaurant_page_js/f6d9d2e69612ad2d78dc3aea398b7f7bccbf2248/dist/index.html.

RoyNyaga replied:
Ok

bolah2009 commented on src/menupage.js:
```diff
@@ -1,8 +1,8 @@
-  }
-
+/* eslint-disable no-mixed-spaces-and-tabs,
+import/prefer-default-export, no-param-reassign, no-tabs */
```
Please, let's export the module using default export instead of ignoring the rule.

adding the following, in this case, should solve the linter warning instead of ignoring it.
```js
export default Menu
```

Also, `npx eslint src --fix` should fixed the `no-mixed-spaces-and-tabs` rule.

bolah2009 commented on src/index.js:
```diff
@@ -1,8 +1,12 @@
+
+const contentDiv = document.querySelector('#content');
+const navLinks = document.querySelectorAll('.nav-link');
 const homeContent = `<div class="container" id="home-section">
```
Could you add this to the homepage module instead? Kindly do the same for other modules.

The contents of each tab are stored in a `module`. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/tab_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/restaurant-page))

bolah2009 commented on package.json:
```diff
@@ -15,7 +17,12 @@
+    "eslint-plugin-import": "^2.19.1",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-react": "^7.17.0",
+    "eslint-plugin-react-hooks": "^1.7.0",
```
These are `React` specific Eslint plugin. I recommend you remove them as they are useless for this project.

bolah2009 reviewed (changes requested):
Good job with the [restaurant page](https://raw.githack.com/RoyNyaga/microverse_restaurant_page_js/f6d9d2e69612ad2d78dc3aea398b7f7bccbf2248/dist/index.html#) :clap:

Kindly see comments below and make changes accordingly.

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

RoyNyaga committed:
added default exports, transfered rendered content into specific modules

RoyNyaga committed:
added life version link to readme file

RoyNyaga commented:
Added life version link on the readme file. Transferred every page HTML snippet into its page module and also used default exports for exporting modules. Removed the lines of code suggested by the previous reviewer from the package.json file. I also want to note that the HTML content in my script files raised a lot of tab and spaces issues which gave me no choice than to exclude the rule since I could not fix that by running npx eslint src --fix.

bellom reviewed (approved):
## Hi @RoyNyaga  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Goodluck in the next project :slightly_smiling_face:_ 

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**
